### ISSUE TITLE
Godot 4 TileMap Crashes: Drawing into tilemap, saving the packed scene, then trying to reopen it crashes the Editor

### ISSUE THREAD

**Issue by mirage3d** [godotengine/godot#72161]
### Godot version

v4.0.beta15.official [4fa6edc88]

### System information

Windows 11, Vulkan, NVIDIA GeForce RTX 3050 Ti Laptop GPU

### Issue description

Godot Editor crashes after changing a tilemap inside a packed scene and reopening it, or another copy of it.

I have a workflow where I have a tilemap inside a packed scene to act as a prefab. To create different versions of the prefab, I duplicate the last packed scene, change its name, double-click it to edit the included tilemap, then rename the root node, and then close it. I then repeat the process to create a new prefab.

The first time I double-click to edit the new duplicate, it doesn't open. If I try a second time, the packed scene opens. I can rename the root node, but the Editor feels sluggish. I only have a couple of seconds more before the Editor freezes for a short moment and then exits.

I ran Godot in a separate console by calling "Godot_v4.0-beta15_win64_console" to catch any error messages, and this is what I get:
```
Godot Engine v4.0.beta15.official.4fa6edc88 - https://godotengine.org
Vulkan API 1.3.224 - Using Vulkan Device #0: NVIDIA - NVIDIA GeForce RTX 3050 Ti Laptop GPU

ERROR: TileSetAtlasSource has no tile at (-1818651808, 359).
   at: (scene/resources/tile_set.cpp:4169)
ERROR: TileSetAtlasSource has no tile at (-1818651808, 359).
   at: (scene/resources/tile_set.cpp:4271)
ERROR: TileSetAtlasSource has no tile at (-1818651808, 359).
   at: (scene/resources/tile_set.cpp:4286)
ERROR: Condition "!p_tile_set.is_valid()" is true.
   at: draw_tile (scene/2d/tile_map.cpp:1351)
ERROR: TileSetAtlasSource has no tile at (-1818651808, 359).
   at: (scene/resources/tile_set.cpp:4169)
```
Sometimes I only get these 2 errors:
```
ERROR: Condition "!p_tile_set.is_valid()" is true.
   at: edit (editor/plugins/tiles/tile_set_atlas_source_editor.cpp:123)
ERROR: Condition "!p_tile_set" is true.
   at: set_atlas_source (editor/plugins/tiles/tile_atlas_view.cpp:410)
```

### Steps to reproduce

~~I already outlined the process above, and it pretty much summarizes the steps to reproduce this.~~

I posted a couple of videos below to showcase what happens.

### Minimal reproduction project

[Tilemap Crash Bug.zip](https://github.com/godotengine/godot/files/10516658/Tilemap.Crash.Bug.zip)

**Comment by mirage3d** [comment]
Just an update as this is occurring frequently and becoming an issue as I have lots of similar assets that I need to edit from time to time.

I noticed this error message in the console, which acts as a warning that any further handling of TileMap nodes will result in a crash:

```
ERROR: Condition "p_resource.is_null()" is true. Returning: false
   at: is_resource_read_only (editor/editor_node.cpp:4084)

```

I don't want to speculate, but first time, and sometimes couple of times, I want to edit packed scenes with embedded TileMaps, everything works fine. If I try to edit another packed scene, the editor freezes momentarily then crashes.
Another effect happens, that might or might not be related to this issue, is that after working ok the first time, attempting to select tiles in the TileMap to move them gives me black tiles, then the tiles disappear completely.

Here's a screen recording of a crash:
https://user-images.githubusercontent.com/56555/215395064-e951d36d-73d5-4da6-8a3b-5a0800d22235.mp4

And another showing that the file opens no problems the first time. After closing it, the editor immediately freezes and crashes:
https://user-images.githubusercontent.com/56555/215395508-13750180-f883-44d2-a670-7bc21e3cac9e.mp4

**Comment by KoBeWi** [comment]
This was fixed by #73259
